lib/semantics/expression.cc

```diff
@@ -1888,8 +1888,17 @@ MaybeExpr ExpressionAnalyzer::Analyze(const parser::Expr &expr) {
 }
 
 MaybeExpr ExpressionAnalyzer::Analyze(const parser::Variable &variable) {
-  FixMisparsedFunctionReference(context_, variable.u);
-  return Analyze(variable.u);
+  if (variable.typedExpr) {
```

Comment:
Since this is pretty much what's done for expressions, maybe it could become a subroutine.

Reply:
OK, I've made this change.